src/runtime/internal/dev.ts

```diff
@@ -256,11 +256,47 @@ export class SvelteComponentTyped<
 	 */
 	$$slot_def: Slots;
 
-	constructor(options: IComponentOptions<Props>) {
+	constructor(options: ComponentConstructorParams<Props>) {
 		super(options);
 	}
 }
 
+/**
+ * Convenience type to get the type of a Svelte component. Useful for example in combination with
+ * dynamic components and `<svelte:element>`.
+ * 
+ * Example:
+ * ```html
+ * <script lang="ts">
+ * 	import type { ComponentType, SvelteComponentTyped } from 'svelte';
+ * 	import Component1 from './Component1.svelte';
+ * 	import Component2 from './Component2.svelte';
+ * 
+ * 	const component: ComponentType = someLogic() ? Component1 : Component2;
+ * 	const componentOfCertainSubType: ComponentType<SvelteComponentTyped<{ needsThisProp: string }>> = someLogic() ? Component1 : Component2;
+ * </script>
+ * 
+ * <svelte:element this={component} />
+ * <svelte:element this={componentOfCertainSubType} needsThisProp="hello" />
+ * ```
+ */
+export type ComponentType<T extends SvelteComponentTyped = SvelteComponentTyped<any, any, any>> =
+	new (p: ComponentConstructorParams<T extends SvelteComponentTyped<infer X, any, any> ? X : any>) => T;
```

Comment:
I don't quite get why we need the generic to extend from `SvelteComponentTyped`, that makes this convenience type depends on the user also importing `SvelteComponentTyped`, otherwise the props will always be typed as `any`.

I haven't looked too deeply here, mostly because I can't seem to get this working locally (presuming it needs to be supported on the language-tools side as well), but couldn't this be [snippet below] and still behave the same?
```ts
export type ComponentType<T extends Props = Props> = new (p: ComponentConstructorParams<Props>) => T;
```

Reply:
Could you post the code that doesn't work for you? I didn't change anything in language tools with regards to this PR

Reply:
Ah I now see that it works (and its errors), the additional import and usage of `SvelteComponentTyped` is still a bummer though.

Reply:
It was exactly the same as the one written in the docs (different imported components), apologies should've mentioned it sooner.

Reply:
I see. This was actually the initial implementation (see first commit), but I decided against it because I feel that this approach serves more use cases. For example you could do `ComponentType<Another imported component>`.